packages/ckeditor5-dev-tests/bin/save-revision.js

```diff
@@ -21,24 +21,31 @@ const mainRepoUrl = 'https://github.com/ckeditor/ckeditor5';
 const revisionBranch = `${ branch }-revisions`;
 
 // Clone the repository.
-exec( `git clone -b ${ revisionBranch } ${ mainRepoUrl }.git` );
+exec( `git clone ${ mainRepoUrl }.git` );
+
+// And check out to the revision branch.
+exec( `git checkout ${ revisionBranch } ` );
 
 // Change current dir to cloned repository.
 process.chdir( path.join( process.cwd(), 'ckeditor5' ) );
 
 // Install Mgit.
 exec( 'npm install -g mgit2' );
 
-// Get the latest `mgit.json`.
-exec( `git checkout origin/${ branch } .` );
+// Sync the revision branch with the master.
+exec( `git checkout ${ branch } -- .` );
+
+// Remove all files from "staged".
+exec( 'git reset -q .' );
```

Comment:
I think you may sometime need to pass additional options to also clean up old directories.

Reply:
Not sure but this command does what you wrote here - https://github.com/ckeditor/ckeditor5-dev/issues/375#issuecomment-376487447.